specs/DeliverInputMessagesToHostMessageQueue.md

```diff
@@ -0,0 +1,101 @@
+# Background
+WebView2 .NET developers often encounter issues where certain default events are not raised in the WebView2 .NET control. 
+
+This issue occurs because user inputs are directly delivered to the browser, and the host application does not receive the corresponding messages. 
+
+This causes some default event handling API of .NET control not to work, such as PreProcessMessage and ProcessCmdKey in WinForms.
+
+It also prevents some keys from reaching Key Event realated APIs such as OnKeyDown and ToolStripMenuItem.
+
+In order to solve this type of issue, we provide an api which allows user inputs pass through the browser to the host app.
+
+# Description
+`DeliverInputMessagesToHostMessageQueue` allows user input messages to pass through the browser window to be received by an app process window.
+
+The messages can be received by Win32 API ::GetMessage() or ::PeekMessage(). This provides the host app a chance to handle the message before dispatching to the WebView2 HWND.
+
+If the host app does not handle input, it is forwarded to the browser process on the user's behalf. This API does not introduce any requirement for the developer to forward all input as is the case with visual hosting. This API should not be used with visual hosting, and has no effect when using  CreateCoreWebView2CompositionControllerWithOptions to create the controller.
+
+# Examples
+## Win32 C++
+```cpp
+HRESULT AppWindow::CreateControllerWithInputPassthrough()
+{
+      //! [CreateControllerWithOptions]
+    auto webViewEnvironment10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+    if (!webViewEnvironment10)
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    wil::com_ptr<ICoreWebView2ControllerOptions> options;
+    CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2ControllerOptions(options.GetAddressOf()));
+
+    wil::com_ptr<ICoreWebView2ControllerOptions3> webView2ControllerOptions3;
+    if (SUCCEEDED(options->QueryInterface(IID_PPV_ARGS(&webView2ControllerOptions3))))
+    {
+        CHECK_FAILURE(webView2ControllerOptions3->put_DeliverInputMessagesToHostMessageQueue(TRUE));
```

Comment:
This property is named to sound like a method (starts with a verb). Trying to think of alternatives ...

Reply:
This is an Options class, and we are generally okay with "methody properties" on Options classes, since they are really just method parameters packaged into a class. Plenty of precedence like

```
runtimeclass AddPackageOptions
{
    Boolean AllowUnsigned { get; set; };
    Boolean DeferRegitrationWhenPackagesAreInUse { get; set; };
    Boolean ForceTargetAppShutdown { get; set; };
    Boolean InstallAllResources { get; set; };
}

runtimeclass ContactQueryOptions
{
    Boolean IncludeContactsFromHiddenLists { get; set; };
}
```

Reply:
Alternate name could be `EnableHostInputProcessing` rather than `DeliverInputMessagesToHostMessageQueue`.

"Enable" is a stronger term than "Allow". ("Allow" provides an option that may not be exercised. "Enable" is a requirement.)

Reply:
I like `EnableHostInputProcessing` too in that it describes the feature more than the implementation detail.

Reply:
Sounds good. Let's use `EnableHostInputProcessing`. Thanks!